subsys/bluetooth/audio/ascs.c

```diff
@@ -2071,6 +2098,60 @@ static bool ascs_parse_metadata(struct bt_data *data, void *user_data)
 				}
 			}
 		}
+
+		break;
+	}
+	case BT_AUDIO_METADATA_TYPE_PARENTAL_RATING:
+		if (data_len != 1) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_INVALID,
+						       data_type);
+			result->err = -EBADMSG;
+			return false;
+		}
+
+		break;
+	case BT_AUDIO_METADATA_TYPE_AUDIO_STATE: {
+		uint8_t state;
+
+		if (data_len != sizeof(state)) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_INVALID,
+						       data_type);
+			result->err = -EBADMSG;
+			return false;
+		}
+
+		state = data_value[0];
+		if (state != BT_AUDIO_ACTIVE_STATE_DISABLED &&
+		    state != BT_AUDIO_ACTIVE_STATE_ENABLED) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED,
+						       data_type);
+			result->err = -EINVAL;
+			return false;
+		}
+
+		break;
+	}
+	/* TODO: Consider rejecting BT_AUDIO_METADATA_TYPE_BROADCAST_IMMEDIATE type */
+	case BT_AUDIO_METADATA_TYPE_BROADCAST_IMMEDIATE:
+		if (data_len != 0) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_INVALID,
+						       data_type);
+			result->err = -EBADMSG;
+			return false;
+		}
+
+		break;
+	case BT_AUDIO_METADATA_TYPE_EXTENDED:
+	case BT_AUDIO_METADATA_TYPE_VENDOR:
+	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO:
+	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO_URI:
+		break;
+	default:
+		LOG_WRN("Unsupported metadata type 0x%02x", data_type);
+		*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_UNSUPPORTED,
+					       data_type);
+		result->err = -ENOTSUP;
+		return false;
```

Comment:
The conclusion we had, before this change, was that the application would be responsible for rejecting invalid Assigned Numbers values. I think we should keep that part (i.e. leave it up to the application to determine whether a type is valid or not) as new context types may be added and can be updated in the application, without requiring a Zephyr update).

Reply:
As per TS, the error code can be Rejected Metadata or Invalid Metadata